s30566-Daniel_Makowski/2.Uprawnienia Dostępu/Zadania.txt

```diff
@@ -0,0 +1,109 @@
+
+	=======================ODPOWIEDŹ=======================
+
+	chmod 000 ./
+
+- usuwając wszelkie uprawnienia dostępu do katalogu nadrzędnego (nadkatalogu).
+
+	=======================ODPOWIEDŹ=======================
+
+	chmod 000 ../
+
+	- W jakich przypadkach możemy wykonać wtedy polecenie cd ? 
+	
+	=======================ODPOWIEDŹ=======================
+		Jedynie w przypadku ścieżki bezwzględnej do miejsca do którego mamy prawo (zakładając, że nie mamy dostępu do roota ani praw sudo).
+
+	- W jakich przypadkach możemy wykonać polecenie chmod ? 
+	
+	=======================ODPOWIEDŹ=======================
+		Jedynie jeżeli katalog należy do nas, bądź mamy dostęp do roota lub sudo.
+
+	- Czy możemy bezpośrednio przeskoczyć do katalogu ABC/XYZ, jeśli nie mamy uprawnienia wstępu do ABC, ale mamy do XYZ ?
+	
+	=======================ODPOWIEDŹ=======================
+		Nie możemy.
+
+	- Czy możemy także wrócić korzystając z polecnia cd - ?
+	
+	=======================ODPOWIEDŹ=======================
+		Nie, nie możemy
+
+3. W zespołach 2- lub 3-osobowych (w opisie zadania na githubie proszę umieścić login osób z zespołu) wypróbować możliwość komunikacji przez współdzielony plik: na jednym z kont w zespole utworzyć pusty plik i przydzielić odpowiednie uprawnienia dostępu (do pliku i do katalogu domowego). Wpisywać i odczytywać komunikaty przy użyciu poleceń:
+
+- echo treść_komunikatu > plik
+
+- cat plik
+	Sprawdzić, jaki skutek powoduje zamiana operatora > na operator >> w poleceniu echo. Uruchom także drugi terminal i wykonaj w nim komendę tail -f plik i powtórz powyższe ćwiczenie w pierwszym terminalu (komunikacja za pomocą pliku).
+	
+	=======================ODPOWIEDŹ=======================
+	
+	Używając operatora ">" plik jest nadpisywany, kiedy przy użyciu ">>" Nowa treść jest dopisywana do końca.
+	Polecenie tail -f pokazuje "na żywo" to co zostało dopisane. Robiłem przy pomocy s29986
+
+4. Znaleźć w swoim katalogu domowym podkatalog public_html (jeśli go nie ma, to utworzyć; musi się on nazywać DOKŁADNIE tak jak podano, pisane małymi literami z podkreśleniem zamiast spacji pomiędzy słowami). Umieścić w nim plik o nazwie strona.html o następującej zawartości:
+
+<HTML>
+
+<BODY>
+
+<H1>To jest moja strona domowa</H1>
+
+</BODY>
+
+</HTML>
+
+Sprawdzić, jakie są minimalne uprawnienia dostępu, które trzeba przydzielić do:
+
+katalogu domowego;
+
+katalogu public_html;
+
+pliku strona.html, aby zawartość pliku była widoczna w przeglądarce internetowej pod adresem http://szuflandia.pjwstk.edu.pl/~twój_login/strona.html
+
+	=======================ODPOWIEDŹ=======================
+
+	Wystarczą dla katalogu domowego 701 folderu public_html (pewnie wystarczyłoby 001 ale wolę nie zmieniać swoich uprawnień dla swojego home directory), dla public_html uprawnienia 001, dla pliku wystarczy już 004. (Dzięki pracy w hostingu wiem, że najlepiej dać 755 na podkatalogi w public_html i 644 na pliki, wówczas bardziej zaawansowane aplikacje mogą działać.)
```

Comment:
Nie poczyniałbym tak daleko idących założeń, że to faktycznie `lepiej`.
Wedle podejścia "zero trust" oraz "least permission" uprawnienia powinny być nadawane tak by nie wykraczały poza to co jest absolutnie wymagane przez aplikacje.
Drugą rzeczą jest to, że w przypadku aplikacji tworzy sie dedykowane "identity" na których aplikacja pracuje i wtedy operujemy na uprawnieniach właściciela. Są miejsca i momenty kiedy celowo dostęp powinien być odebrany np. do czytania logów albo do logów serwera. Aplikacja powinna być w stanie tam pisać na podstawie tego jak serwer aplikacyjny jest skonfigurowany i uruchomiony ale nie powinna być w stanie czytać tych logów bo nie jest to wymagane do uruchomienia samej aplikacji a ponadto często w logach znajdują się dane które mogą umożliwić pozyskanie większej ilości informacji o uprawnieniach jak i konfiguracji serwera aplikacyjnego czy serwera z punktu widzenia systemu operacyjnego.
Tak samo jak nazwę i wersję serwera webowego powinno się usunąć z nagłówków a robi się to przez skonfigurowanie go poprawnie przed rozpoczęciem hostowania serwisów.

Zatem wracając do początku: Nie poczyniałbym tak daleko idących założeń, że "najlepiej dać 755... i 644...".
Tu wszystko opiera się o świadome zarządzanie uprawnieniami dostępu a nie robieniem wg szablonu.
Szablon wszystko upraszcza ale nie powoduje poprawy bezpieczeństwa.

Prosze przemyśl te słowa i gdy wyciągniesz z tego jakieś wnioski popraw swoją wypowiedź.